e2e-tests/tests/otel_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package e2etests
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	otelcollector "github.com/DataDog/opentelemetry-collector-contrib/e2e-tests/otel-collector"
+	"github.com/DataDog/opentelemetry-collector-contrib/e2e-tests/otel-collector/otelparams"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: Can be improved. On datadog-agent we use python code to run the e2e tests and pass the docker secret as an env variable
+var dockerSecret *string = flag.String("docker_secret", "", "Docker secret to use for the tests")
+
+type otelSuite struct {
+	e2e.BaseSuite[otelcollector.Kubernetes]
+}
+
+// TestVMSuite runs tests for the VM interface to ensure its implementation is correct.
+func TestVMSuite(t *testing.T) {
+	extraParams := runner.ConfigMap{}
+	extraParams.Set("ddagent:imagePullRegistry", "669783387624.dkr.ecr.us-east-1.amazonaws.com", false)
+	extraParams.Set("ddagent:imagePullPassword", *dockerSecret, true)
+	extraParams.Set("ddagent:imagePullUsername", "AWS", false)
+
+	otelOptions := []otelparams.Option{}
+	// Use image built in the CI
+	pipelineID, ok1 := os.LookupEnv("E2E_PIPELINE_ID")
+	commitSHA, ok2 := os.LookupEnv("E2E_COMMIT_SHORT_SHA")
+	if ok1 && ok2 {
+		values := fmt.Sprintf(`
+image:
+  tag: %s-%s
+`, pipelineID, commitSHA)
+		otelOptions = append(otelOptions, otelparams.WithHelmValues(values))
+	}
+
+	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(otelcollector.Provisioner(otelcollector.WithOTelOptions(otelOptions...), otelcollector.WithExtraParams(extraParams)))}
```

Comment:
It would be great to have the same option here as the OTel Agent `WithOTelConfig` to pass in a go embed collector config yaml if possible

Reply:
Just added that options and edited the test to have an example about how to use it!